### PR TITLE
WIP [1/2]: ENG-918 Implement Artifact Details Page

### PR DESCRIPTION
**Pull request by agiron123**
Opening up as a draft PR for some self code review. Loom demo to come soon :)

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: examples/churn_prediction/Build and Deploy Churn Ensemble.ipynb [hunk 1]
```diff
@@ -36,7 +36,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": 6,
```
Comment (agiron123) [inline examples/churn_prediction/Build and Deploy Churn Ensemble.ipynb:39]: Need to revert this file, these always show up in VC when I update :)

File: src/ui/app/index.tsx [hunk 2]
```diff
@@ -31,22 +31,23 @@ const App = () => {
...
       <Route path={`/${pathPrefix}/integrations`} element={<RequireAuth user={user}><IntegrationsPage user={user} /> </RequireAuth>} />
       <Route path={`/${pathPrefix}/integration/:id`} element={<RequireAuth user={user}><IntegrationDetailsPage user={user} /> </RequireAuth>} />
       <Route path={`/${pathPrefix}/workflows`} element={<RequireAuth user={user}><WorkflowsPage user={user} /> </RequireAuth>} />
-      <Route path={`/${pathPrefix}/login`} element={ user && user.apiKey ? <Navigate to="/" replace /> : <LoginPage />} />
+      <Route path={`/${pathPrefix}/login`} element={user && user.apiKey ? <Navigate to="/" replace /> : <LoginPage />} />
       <Route path={`/${pathPrefix}/account`} element={<RequireAuth user={user}><AccountPage user={user} /> </RequireAuth>} />
       <Route path={`/${pathPrefix}/workflow/:id`} element={<RequireAuth user={user}><WorkflowPage user={user} /> </RequireAuth>} />
+      <Route path={`/${pathPrefix}/workflow/:workflowId/result/:workflowDagResultId/artifact/:artifactId`} element={<RequireAuth user={user}><ArtifactDetailsPage user={user} /> </RequireAuth>} />
```
Comment (agiron123) [inline src/ui/app/index.tsx:42]: Not 100% sure about the routing here. Open to feedback and happy to update the route used here.

I do, however, know that we need to have the workflowResultId and artifactResultId on hand to be able to get the data that we need for this Artifact details page.

Technically speaking, this is more of an ArtifactResults detail page.

File: src/ui/common/src/components/pages/artifact/id/index.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+type ArtifactDetailsHeaderProps = {
+    artifactName: string;
+    createdAt?: string;
+    sourceLocation?: string;
+};
+
+const ArtifactDetailsHeader: React.FC<ArtifactDetailsHeaderProps> = ({
+    artifactName,
+    // TODO: add these back once we have support for getting createdAt and sourceLocation.
```
Comment (agiron123) [inline src/ui/common/src/components/pages/artifact/id/index.tsx:29]: All we really have here is the name of the artifact result.

Don't really even have a status here either, so I'm just showing the check icon for now.

Happy to remove the check icon if we find that the status indicator here is meaningless for an artifact result.

File: src/ui/common/src/components/pages/artifact/id/index.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                            EXPORT
+                        </Button>
+                    </Box>
+                    <Box width="100%" marginTop="12px">
+                        <Typography variant="h5" component="div" marginBottom="8px">
+                            Preview
+                        </Typography>
+                        <StickyHeaderTable data={parsedData} />
+                    </Box>
+                    {/* <Box display="flex" width="100%" paddingTop="40px">
```
Comment (agiron123) [inline src/ui/common/src/components/pages/artifact/id/index.tsx:115]: The KeyValueTable component here is meant to be a component to easily display the results of metrics and checks. 

I'm not exactly sure how to go about getting the associated metrics and Checks for an ArtifactResult. Could use a little help here.

File: src/ui/common/src/components/pages/artifact/id/index.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        );
+    }
+
+    const parsedData = JSON.parse(artifactResult.result.data);
+
+    return (
+        <Layout user={user}>
+            <Box width={'800px'}>
+                <Box width="100%">
+                    <Box width="100%" display="flex">
```
Comment (agiron123) [inline src/ui/common/src/components/pages/artifact/id/index.tsx:101]: Going to take this header and make into more of a common component in the future. This header is used across our new mocks.

File: src/ui/app/index.tsx (hunk 2 above)
Comment (agiron123) [reply src/ui/app/index.tsx:42]: Discussed this route some more with @likawind and outlined some possible solutions here:
https://www.notion.so/aqueducthq/UI-Redesign-API-Routes-ed8b1f5945474683a78c6ae4bf5cfc7e

TLDR: we're going to need the workflowId and workflowDagResultId as part of the route to be able to drill down into the artifact that we want. Whether we implement new routes for fetching the artifact in one go is still TBD. In the interim at least, we can fetch the whole workflow and traverse it's dag to get to the data that we need.

File: src/ui/common/src/reducers/workflow.ts [hunk 6]
```diff
@@ -128,6 +128,7 @@ export const handleGetArtifactResults = createAsyncThunk<
     );
 
     const body = await res.json();
+    console.log('handleGetArtifactResults body: ', body);
```
Comment (agiron123) [inline src/ui/common/src/reducers/workflow.ts:131]: remove log

**Comment by agiron123** [comment]
Closing in favor of #351
